File: Baekjoon/Class2/08.1259 팰린드롬수.py
import sys
input = sys.stdin.readline

'''
def is_palindrome(string):
    length = len(string)
    for i in range(length // 2): #절반까지 비교
        if string[i] != string[length - 1 - i]:
            return False
    return True
'''

def main():
    while True:
        string = input().strip()
        if string == "0":
            break
        #if(is_palindrome(string)):
        if(string == string[::-1]):
            print("yes")
        else:
            print("no")

main()

File: Baekjoon/Class2/08.1259 팰린드롬수.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").split("\n");

function isPalindrome(str) {
    const len = str.length;
    for(let i = 0; i < len / 2; i++) // 절반까지 비교
        if (str.charAt(i) !== str.charAt(len - 1 - i)) return false;
    return true;
}

let cursor = 0;
let out = "";
while(true) {
    const str = input[cursor++].trim();
    if(str === "0") break;
    // if(str === str.split("").reverse().join(""))
    if(isPalindrome(str))
        out += "yes\n";
    else
        out += "no\n";
}
console.log(out)